app/scripts/controllers/authentication/authentication-controller.ts

```diff
@@ -0,0 +1,266 @@
+import {
+  BaseController,
+  RestrictedControllerMessenger,
+  StateMetadata,
+} from '@metamask/base-controller';
+import { HandleSnapRequest } from '@metamask/snaps-controllers';
+import {
+  createSnapPublicKeyRequest,
+  createSnapSignMessageRequest,
+} from './auth-snap-requests';
+import {
+  createLoginRawMessage,
+  getAccessToken,
+  getNonce,
+  login,
+} from './services';
+
+const THIRTY_MIN_MS = 1000 * 60 * 30;
```

Comment:
`@metamask/utils` has a nice utility function for this. Something like:

```
inMilliseconds(30, Duration.Minute);
```

Reply:
I will promise to do this in the next PR - been itching to get this merged 😛